pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go

```diff
@@ -0,0 +1,159 @@
+//  Copyright 2021 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package watchers
+
+import (
+	"github.com/cilium/cilium/pkg/k8s"
+	cilium_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	k8sUtils "github.com/cilium/cilium/pkg/k8s/utils"
+
+	"github.com/sirupsen/logrus"
+)
+
+type CEBSubcriber struct {
+	kWatcher *K8sWatcher
+}
+
+func newCEBSubscriber(k *K8sWatcher) *CEBSubcriber {
+	return &CEBSubcriber{
+		kWatcher: k,
+	}
+}
+
+func (cs *CEBSubcriber) OnAdd(obj interface{}) {
+	ceb, _ := obj.(*cilium_v2a1.CiliumEndpointBatch)
+	cs.CEBAdded(ceb)
+}
```

Comment:
A new subscriber should be in it's own package rather than as part of the K8s watcher (that is the point of the refactor). If there already exists a package for CEB, then this code belongs there. 

Additionally, we can define our own "subscriber" by defining a new method set (inside `pkg/k8s/watchers/subscriber/ceb.go` for ex.) for `CiliumEndpointBatch` objects, rather than using the "raw" subscriber that forces us to type assert due to the methods accepting only `interface{}`. See `pkg/k8s/watchers/subscriber/service.go`.

Reply:
Cilium-agent doesn't use the actual `CEB` package, it watches for the `CEB`s and calls the `CEP` handler. There is nothing more we do in `cilium-agent`. The actual CEB logic resides in the operator. hence, this code is part of `k8swatcher`.

Main reason I had skipped having a separate `ceb subscriber` is just a replica of `pkg/k8s/watchers/subscriber/raw.go`.I prefer having lesser code, rather than having duplicate code.

Reply:
Updated with separate CEB subscriber.